### ISSUE TITLE
Whispers of the Raven (10607) (BB #947)

### ISSUE THREAD

**Issue by OregonAsari** [OregonCore/OregonCore#947]
This issue was migrated from bitbucket.
**Original Reporter:** bewtihozzy
**Original Date:** 31.05.2015 17:56:23 GMT+0000
**Original Priority:** major
**Original Type:** bug
**Original State:** resolved
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/947

<hr>

Whispers of the Raven (10607)

Kill Grishna arakkoa (22232)  to gain Understanding Ravenspeech for 2 minutes. While under the effects of the spell, you can find and receive the prophecy from the four shrines. Return when you have visited all four.
Receive the First Prophecy
Receive the Second Prophecy
Receive the Third Prophecy
Receive the Fourth Prophecy

bug: When you kill the Grishna arakkoa, it doesnt give you the buff.

-> https://youtu.be/SNMBbDA-nR8

Edit: I moved the mob to that spot so he doesnt get attacked by guards xD

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** bewtihozzy
**Original Date:** 31.05.2015 17:57:44 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/947/#comment-18491278

<hr>

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Selphius
**Original Date:** 22.11.2015 18:38:09 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/947/#comment-23498655

<hr>

Fix for this:

```
#!c++

/*######
## npc_grishna_arakkoa
######*/

struct npc_grishna_arakkoaAI : public ScriptedAI
{
    npc_grishna_arakkoaAI(Creature *c) : ScriptedAI(c) {}

    void Reset() { }

    void JustDied(Unit* killer)
    {
        if (Player *plWho = killer->GetCharmerOrOwnerPlayerOrPlayerItself())
        {
            if (plWho->GetQuestStatus(10607) == QUEST_STATUS_INCOMPLETE)
                if (urand(0, 2))
                    plWho->CastSpell(plWho, 37642, true, 0, 0, 0);
        }
    }

    void UpdateAI(const uint32 /*diff*/)
    {
        if (!UpdateVictim())
            return;
        DoMeleeAttackIfReady();
    }
};

CreatureAI* GetAI_npc_grishna_arakkoa(Creature* pCreature)
{
    return new npc_grishna_arakkoaAI(pCreature);
}

newscript = new Script;
    newscript->Name = "npc_grishna_arakkoa";
    newscript->GetAI = &GetAI_npc_grishna_arakkoa;
    newscript->RegisterSelf();
```

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** AlphafoxBHN
**Original Date:** 22.11.2015 20:03:27 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/947/#comment-23499654

<hr>

i have compile errors .... after change to this 

```
#!c++

#include "ScriptPCH.h"
#include "ScriptMgr.h"
#include "ScriptedCreature.h"
#include "ScriptedGossip.h"
#include <cstring>
/*######
## npc_grishna_arakkoa
######*/

struct npc_grishna_arakkoaAI : public ScriptedAI
{
    npc_grishna_arakkoaAI(Creature *c) : ScriptedAI(c) {}

    void Reset() { }

    void JustDied(Unit* killer)
    {
        if (Player *plWho = killer->GetCharmerOrOwnerPlayerOrPlayerItself())
        {
            if (plWho->GetQuestStatus(10607) == QUEST_STATUS_INCOMPLETE)
                if (urand(0, 2))
                    plWho->CastSpell(plWho, 37642, true, 0, 0, 0);
        }
    }

    void UpdateAI(const uint32 /*diff*/)
    {
        if (!UpdateVictim())
            return;
        DoMeleeAttackIfReady();
    }
};

CreatureAI* GetAI_npc_grishna_arakkoa(Creature* pCreature)
{
    return new npc_grishna_arakkoaAI(pCreature);
}

void AddSC_npc_grishna_arakkoa()
{
    Script* newscript;

   newscript = new Script;
    newscript->Name = "npc_grishna_arakkoa";
    newscript->GetAI = &GetAI_npc_grishna_arakkoa;
    newscript->RegisterSelf();
}


```

no compile error

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Selphius
**Original Date:** 22.11.2015 20:21:22 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/947/#comment-23499852

<hr>

Well you will have compile errors since you "create entire new script" from what I see by 

```
#!c++
void AddSC_npc_grishna_arakkoa()

```

Since this quest is in Blade Edge Mountains, you could add it to Blade Edge Mountains.cpp where rest of the quest for this zone are located, you wouldn't have to add all those new lines like you did.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** AlphafoxBHN
**Original Date:** 22.11.2015 20:23:15 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/947/#comment-23499865

<hr>

Ah okay than i have do some mistakes :D

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** Selphius
**Original Date:** 02.12.2015 22:56:56 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/947/#comment-23762788

<hr>

Issue #988 was marked as a duplicate of this issue.

**Comment by OregonAsari** [comment]
This comment was migrated from bitbucket.
**Original User:** LordUsagi
**Original Date:** 25.12.2015 00:51:37 GMT+0000
**Direct Link:** https://bitbucket.org/oregon/oregoncore/issues/947/#comment-24313406

<hr>

[DB/SAI] Script for quest Whispers of the Raven God.
         Fixes #947.

→ <<cset ca5dda4f034b>>
